### ISSUE TITLE
markTestIncomplete causes "Nesting level too deep - recursive dependency?"

### ISSUE THREAD

**Issue by RyanHavoc** [Codeception/Codeception#2748]
I'm running Codeception 2.1.5 with PHPUnit 4.8.21 against Laravel 5.2 and when I have two consecutive tests with $this->markTestIncomplete(); I get the following error after the second test:

```
PHP Fatal error:  Nesting level too deep - recursive dependency? in /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/PHPUnit/Listener.php on line 55
PHP Stack trace:
PHP   1. {main}() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/codecept:0
PHP   2. Symfony\Component\Console\Application->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/codecept:28
PHP   3. Symfony\Component\Console\Application->doRun() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/symfony/console/Application.php:117
PHP   4. Symfony\Component\Console\Application->doRunCommand() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/symfony/console/Application.php:186
PHP   5. Symfony\Component\Console\Command\Command->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/symfony/console/Application.php:787
PHP   6. Codeception\Command\Run->execute() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/symfony/console/Command/Command.php:256
PHP   7. Codeception\Codecept->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/Command/Run.php:184
PHP   8. Codeception\Codecept->runSuite() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/Codecept.php:172
PHP   9. Codeception\SuiteManager->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/Codecept.php:203
PHP  10. Codeception\PHPUnit\Runner->doEnhancedRun() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/SuiteManager.php:153
PHP  11. PHPUnit_Framework_TestSuite->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/PHPUnit/Runner.php:89
PHP  12. PHPUnit_Framework_TestCase->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/phpunit/phpunit/src/Framework/TestSuite.php:747
PHP  13. PHPUnit_Framework_TestResult->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/phpunit/phpunit/src/Framework/TestCase.php:724
PHP  14. PHPUnit_Framework_TestResult->addFailure() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/phpunit/phpunit/src/Framework/TestResult.php:698
PHP  15. Codeception\PHPUnit\Listener->addIncompleteTest() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/phpunit/phpunit/src/Framework/TestResult.php:277
PHP  16. in_array() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/PHPUnit/Listener.php:55

Fatal error: Nesting level too deep - recursive dependency? in /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/PHPUnit/Listener.php on line 55

Call Stack:
    0.0002     362984   1. {main}() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/codecept:0
    0.1952    3927200   2. Symfony\Component\Console\Application->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/codecept:28
    0.2121    4213792   3. Symfony\Component\Console\Application->doRun() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/symfony/console/Application.php:117
    0.2122    4213792   4. Symfony\Component\Console\Application->doRunCommand() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/symfony/console/Application.php:186
    0.2123    4213792   5. Symfony\Component\Console\Command\Command->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/symfony/console/Application.php:787
    0.2125    4215920   6. Codeception\Command\Run->execute() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/symfony/console/Command/Command.php:256
    3.5413    7165416   7. Codeception\Codecept->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/Command/Run.php:184
    3.5448    7168480   8. Codeception\Codecept->runSuite() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/Codecept.php:172
    3.6597    9035600   9. Codeception\SuiteManager->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/Codecept.php:203
    3.6739    9056928  10. Codeception\PHPUnit\Runner->doEnhancedRun() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/SuiteManager.php:153
    3.6792    9070464  11. PHPUnit_Framework_TestSuite->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/PHPUnit/Runner.php:89
   24.0615   30563640  12. PHPUnit_Framework_TestCase->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/phpunit/phpunit/src/Framework/TestSuite.php:747
   24.0616   30563640  13. PHPUnit_Framework_TestResult->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/phpunit/phpunit/src/Framework/TestCase.php:724
   25.3630   31093488  14. PHPUnit_Framework_TestResult->addFailure() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/phpunit/phpunit/src/Framework/TestResult.php:698
   25.3631   31093904  15. Codeception\PHPUnit\Listener->addIncompleteTest() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/phpunit/phpunit/src/Framework/TestResult.php:277
   25.3631   31093904  16. in_array() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/PHPUnit/Listener.php:55




FATAL ERROR. TESTS NOT FINISHED.
Nesting level too deep - recursive dependency?
in /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/PHPUnit/Listener.php:55


  [Symfony\Component\Debug\Exception\FatalErrorException]
  Nesting level too deep - recursive dependency?


PHP Fatal error:  Uncaught Illuminate\Contracts\Container\BindingResolutionException: Target [Illuminate\Contracts\Debug\ExceptionHandler] is not instantiable. in /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/laravel/framework/src/Illuminate/Container/Container.php:752
Stack trace:
#0 /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/laravel/framework/src/Illuminate/Container/Container.php(633): Illuminate\Container\Container->build('Illuminate\\Cont...', Array)
#1 /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(674): Illuminate\Container\Container->make('Illuminate\\Cont...', Array)
#2 /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(154): Illuminate\Foundation\Application->make('Illuminate\\Cont...')
#3 /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(79): Illuminate\Foundation\Bootstrap\Han in /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/laravel/framework/src/Illuminate/Container/Container.php on line 752
PHP Stack trace:
PHP   1. {main}() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/codecept:0
PHP   2. Symfony\Component\Console\Application->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/codecept:28
PHP   3. Symfony\Component\Console\Application->doRun() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/symfony/console/Application.php:117
PHP   4. Symfony\Component\Console\Application->doRunCommand() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/symfony/console/Application.php:186
PHP   5. Symfony\Component\Console\Command\Command->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/symfony/console/Application.php:787
PHP   6. Codeception\Command\Run->execute() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/symfony/console/Command/Command.php:256
PHP   7. Codeception\Codecept->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/Command/Run.php:184
PHP   8. Codeception\Codecept->runSuite() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/Codecept.php:172
PHP   9. Codeception\SuiteManager->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/Codecept.php:203
PHP  10. Codeception\PHPUnit\Runner->doEnhancedRun() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/SuiteManager.php:153
PHP  11. PHPUnit_Framework_TestSuite->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/PHPUnit/Runner.php:89
PHP  12. PHPUnit_Framework_TestCase->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/phpunit/phpunit/src/Framework/TestSuite.php:747
PHP  13. PHPUnit_Framework_TestResult->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/phpunit/phpunit/src/Framework/TestCase.php:724
PHP  14. PHPUnit_Framework_TestResult->addFailure() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/phpunit/phpunit/src/Framework/TestResult.php:698
PHP  15. Codeception\PHPUnit\Listener->addIncompleteTest() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/phpunit/phpunit/src/Framework/TestResult.php:277
PHP  16. in_array() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/PHPUnit/Listener.php:55

Fatal error: Uncaught Illuminate\Contracts\Container\BindingResolutionException: Target [Illuminate\Contracts\Debug\ExceptionHandler] is not instantiable. in /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/laravel/framework/src/Illuminate/Container/Container.php on line 752

Call Stack:
    0.0002     362984   1. {main}() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/codecept:0
    0.1952    3927200   2. Symfony\Component\Console\Application->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/codecept:28
    0.2121    4213792   3. Symfony\Component\Console\Application->doRun() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/symfony/console/Application.php:117
    0.2122    4213792   4. Symfony\Component\Console\Application->doRunCommand() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/symfony/console/Application.php:186
    0.2123    4213792   5. Symfony\Component\Console\Command\Command->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/symfony/console/Application.php:787
    0.2125    4215920   6. Codeception\Command\Run->execute() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/symfony/console/Command/Command.php:256
    3.5413    7165416   7. Codeception\Codecept->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/Command/Run.php:184
    3.5448    7168480   8. Codeception\Codecept->runSuite() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/Codecept.php:172
    3.6597    9035600   9. Codeception\SuiteManager->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/Codecept.php:203
    3.6739    9056928  10. Codeception\PHPUnit\Runner->doEnhancedRun() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/SuiteManager.php:153
    3.6792    9070464  11. PHPUnit_Framework_TestSuite->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/PHPUnit/Runner.php:89
   24.0615   30563640  12. PHPUnit_Framework_TestCase->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/phpunit/phpunit/src/Framework/TestSuite.php:747
   24.0616   30563640  13. PHPUnit_Framework_TestResult->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/phpunit/phpunit/src/Framework/TestCase.php:724
   25.3630   31093488  14. PHPUnit_Framework_TestResult->addFailure() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/phpunit/phpunit/src/Framework/TestResult.php:698
   25.3631   31093904  15. Codeception\PHPUnit\Listener->addIncompleteTest() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/phpunit/phpunit/src/Framework/TestResult.php:277
   25.3631   31093904  16. in_array() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/PHPUnit/Listener.php:55

Illuminate\Contracts\Container\BindingResolutionException: Target [Illuminate\Contracts\Debug\ExceptionHandler] is not instantiable. in /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/laravel/framework/src/Illuminate/Container/Container.php on line 752

Call Stack:
    0.0002     362984   1. {main}() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/codecept:0
    0.1952    3927200   2. Symfony\Component\Console\Application->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/codecept:28
    0.2121    4213792   3. Symfony\Component\Console\Application->doRun() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/symfony/console/Application.php:117
    0.2122    4213792   4. Symfony\Component\Console\Application->doRunCommand() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/symfony/console/Application.php:186
    0.2123    4213792   5. Symfony\Component\Console\Command\Command->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/symfony/console/Application.php:787
    0.2125    4215920   6. Codeception\Command\Run->execute() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/symfony/console/Command/Command.php:256
    3.5413    7165416   7. Codeception\Codecept->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/Command/Run.php:184
    3.5448    7168480   8. Codeception\Codecept->runSuite() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/Codecept.php:172
    3.6597    9035600   9. Codeception\SuiteManager->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/Codecept.php:203
    3.6739    9056928  10. Codeception\PHPUnit\Runner->doEnhancedRun() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/SuiteManager.php:153
    3.6792    9070464  11. PHPUnit_Framework_TestSuite->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/PHPUnit/Runner.php:89
   24.0615   30563640  12. PHPUnit_Framework_TestCase->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/phpunit/phpunit/src/Framework/TestSuite.php:747
   24.0616   30563640  13. PHPUnit_Framework_TestResult->run() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/phpunit/phpunit/src/Framework/TestCase.php:724
   25.3630   31093488  14. PHPUnit_Framework_TestResult->addFailure() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/phpunit/phpunit/src/Framework/TestResult.php:698
   25.3631   31093904  15. Codeception\PHPUnit\Listener->addIncompleteTest() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/phpunit/phpunit/src/Framework/TestResult.php:277
   25.3631   31093904  16. in_array() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/codeception/codeception/src/Codeception/PHPUnit/Listener.php:55
   25.3975   31145944  17. Illuminate\Foundation\Bootstrap\HandleExceptions->handleShutdown() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php:0
   25.3976   31164112  18. Illuminate\Foundation\Bootstrap\HandleExceptions->handleException() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php:118
   25.3976   31164112  19. Illuminate\Foundation\Bootstrap\HandleExceptions->getExceptionHandler() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php:79
   25.3976   31164112  20. Illuminate\Foundation\Application->make() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php:154
   25.3976   31164168  21. Illuminate\Container\Container->make() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/laravel/framework/src/Illuminate/Foundation/Application.php:674
   25.3976   31164168  22. Illuminate\Container\Container->build() /home/vagrant/Code/motionarray-5.2.loc/laravel/vendor/laravel/framework/src/Illuminate/Container/Container.php:633
```

$this->markTestSkipped() works fine so we're using this as a stop-gap but it's annoying that we can't mark out tests properly.

Has anyone else experienced this problem?

**Comment by DavertMik** [comment]
This is fixed in 2.1 branch and looks like it is a good time to release 2.1.6...

**Comment by RyanHavoc** [comment]
@DavertMik Thanks. Confirmed, I installed 2.1.x-dev and the issues has gone away. When is the official release of 2.1.6 due, do you know?
